orderer/common/channelparticipation/restapi.go

```diff
@@ -51,7 +58,43 @@ func (h *HTTPHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	}
 
 	switch req.Method {
-	case http.MethodGet, http.MethodPost, http.MethodDelete:
+	case http.MethodGet:
+		_, err := negotiateContentType(req) // Only application/json for now
+		if err != nil {
+			h.sendResponseJsonError(resp, http.StatusNotAcceptable, err)
+			return
+		}
+
+		channelID, all, err := detectResource(req.URL.Path)
```

Comment:
this is not the way you should implement this logic, instead of manually extracting parameters you should template REST endpoint, for example:

```
  var channelsEndpoint = /participation/v1/channels/{channelID}
```

then you can use

```
   vars := mux.Vars(req)
   channelID := vars["channelID"]
```

Reply:
just realized we are not using `gorilla/mux` package to route REST requests.

Reply:
True! We have github.com/gorilla already vendored in Fabric... I wan't aware of that. I'll use that.